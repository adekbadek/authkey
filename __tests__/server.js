require('dotenv').config()

const request = require('supertest')

const authkey = require('../src/server.js')

const instance = authkey({
  mailgun: {
    apiKey: process.env.MAILGUN_API_KEY,
    domain: process.env.MAILGUN_DOMAIN,
  },
  productName: process.env.PRODUCT_NAME,
  dbFile: '__tests__/db.json',
})

describe('POST /request/:address', () => {
  it('should respond with 200 to valid address', async () => {
    const response = await request(instance).post('/request/something@mail.com')
    expect(response.statusCode).toBe(200)
  })
  it('should respond with 400 to invalid address', async () => {
    const response = await request(instance).post('/request/whatever')
    expect(response.statusCode).toBe(400)
  })
})
