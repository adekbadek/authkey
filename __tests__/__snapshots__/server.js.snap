// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /request/:address should respond with special message if address exists 1`] = `
Object {
  "message": "a key for this address was already requested, key was resent",
}
`;
